### DISCUSSION TITLE
I have IMC data Downstream analysis

### DISCUSSION THREAD

**Discussion by biobarani** [ImmuneDynamics/Spectre#131] [Help]
Hi everyone,

I have IMC data (cell segmented data from cell profiler) in .csv file. I want to do single-cell analysis and spatial analysis with this data. I'm very new to this field. Could anyone suggest how to convert this data into single-cell file format to continue in Seurat or spectre pipeline. Also how to do spatial analysis? Kindly suggest the best way. Thanks in advance!

**Comment by tomashhurst** [comment]
Hi @biobarani , thanks for reaching out!

If you are interested in trying out the analysis in Spectre, you can read the CSV directly into R -- to get started you could check out on of our intro tutorials (https://immunedynamics.io/spectre/getting-started/) or typical cytometry workflows (https://immunedynamics.io/spectre/cytometry/). Once the workflow feels a little more familiar, you could then look at the spatial workflows: https://immunedynamics.io/spectre/spatial/#analysis. The initial steps are designed to read in TIFF files and the extract the single-cell data, but you can essentially just skip all of that and just read in the CSV directly.

There are some differences with how the data will look in the export from CellProfiler, some of it to do with how the data was scaled -- if you run into any issues just reach out here and we'll try to help!

If you are interested in trying the Seurat approach, you can convert a CSV file into a Seurat object and use one of their spatial analysis approaches (https://satijalab.org/seurat/articles/spatial_vignette_2.html). You can have a look here (https://wiki.centenary.org.au/pages/viewpage.action?pageId=169664357#) for how to use CSV data to make a Seurat object, but there are probably better tutorials out there if you search on google. 

Tom
